src/lib/state/transactions.ts

```diff
@@ -0,0 +1,53 @@
+import { TransactionReceipt, TransactionResponse } from '@ethersproject/abstract-provider'
```

Comment:
NB: I would consider it valid feedback to tell me to re-use the existing data structures from app. This feels simplified to me, but is definitely a duplication of work.